### PR TITLE
add entitlements for tray-electron

### PR DESCRIPTION
**Pull request by anjannath**
all the suggested entitlements mentioned in the electron docs
and the entitlement required to run applescripts are added

### CODE REVIEW

File: packaging/CodeReady Containers.app.entitlements [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<plist version="1.0">
+  <dict>
+    <key>com.apple.security.cs.allow-jit</key>
+    <true/>
+    <key>com.apple.security.cs.debugger</key>
+    <true/>
+    <key>com.apple.security.automation.apple-events</key>
+    <true/>
+  </dict>
+</plist>
```
Comment (anjannath) [inline packaging/CodeReady Containers.app.entitlements:12]: I think this should be in the tray repo, but we can do that once we move the installers to the tray repo

File: packaging/package.sh [hunk 2]
```diff
@@ -14,9 +14,10 @@ function sign() {
   local opts=""
   entitlements="${BASEDIR}/$(basename "$1").entitlements"
   if [ -f "${entitlements}" ]; then
-      opts="--entitlements ${entitlements}"
+      codesign --deep --sign "${CODESIGN_IDENTITY}" --options runtime --force --entitlements "${entitlements}" "$1"
```
Comment (anjannath) [inline packaging/package.sh:17]: this was needed as the entitlement file has a space in the name

**Review by gbraad — APPROVED** [review]

File: packaging/CodeReady Containers.app.entitlements (hunk 1 above)
Comment (gbraad) [reply packaging/CodeReady Containers.app.entitlements:12]: the installer will be in it's own repo... if we swap the tray out for another app, the installer should just only need minor modifcations.

**Comment by anjannath** [comment]
Looks like with these entitlement also notarisation is failing, but we got some useful log this time:

```json
{
  "logFormatVersion": 1,
  "status": "Invalid",
  "statusSummary": "Archive contains critical validation errors",
  "statusCode": 4000,
  "archiveFilename": "crc-installer-macos-1.99.2-2-amd64.zip",
  "uploadDate": "2022-02-28T16:18:28Z",
  "sha256": "6c3eb4272434cd69587b30736d0651992342d734bde7a01b629bb8212133d156",
  "ticketContents": null,
  "issues": [
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libEGL.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libEGL.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libswiftshader_libEGL.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libswiftshader_libEGL.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libvk_swiftshader.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libvk_swiftshader.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libGLESv2.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libGLESv2.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libswiftshader_libGLESv2.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libswiftshader_libGLESv2.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libffmpeg.dylib",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libffmpeg.dylib",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Squirrel.framework/Versions/A/Resources/ShipIt",
      "message": "The binary is not signed.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Squirrel.framework/Versions/A/Resources/ShipIt",
      "message": "The signature does not include a secure timestamp.",
      "docUrl": null,
      "architecture": "x86_64"
    },
    {
      "severity": "error",
      "code": null,
      "path": "crc-installer-macos-1.99.2-2-amd64.zip/crc-macos-amd64.pkg/crc.pkg Contents/Payload/Applications/CodeReady Containers.app/Contents/Frameworks/Squirrel.framework/Versions/A/Resources/ShipIt",
      "message": "The executable does not have the hardened runtime enabled.",
      "docUrl": null,
      "architecture": "x86_64"
    }
  ]
}
```

**Pull request merged**